### ISSUE TITLE
Reload when exiting resource pack selection screen

### ISSUE THREAD

**Issue by Gerrygames** [MrKinau/MyResourcePack#2]
When opening and then closing the resource pack selection screen (ESC -> Options -> Resource Packs...) after a server has (un-)applied a resource pack the resources are reloaded.

**Comment by Gerrygames** [comment]
Hi @MrKinau,

I hope this message finds you well. Firstly, I want to extend my sincere apologies for any inconveniences caused by my initial report regarding the resource pack reload issue. Upon further investigation, I have identified that the problem lies within Fabric itself, rather than the mod in question.

To address this issue, I have developed a mixin that successfully resolves the resource pack reload problem. I understand the importance of maintaining a seamless user experience, and I believe this mixin could be a valuable addition to the mod.

<details>
  <summary>Mixin Code</summary>

  ```import net.minecraft.client.MinecraftClient;
import net.minecraft.client.gui.screen.pack.PackScreen;
import net.minecraft.resource.ResourcePackManager;
import net.minecraft.resource.ResourcePackProfile;
import net.minecraft.text.Text;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

import java.nio.file.Path;
import java.util.ArrayList;
import java.util.function.Consumer;
import java.util.function.Predicate;
import java.util.stream.Collectors;

@Mixin(PackScreen.class)
public class MixinPackScreen {

	@Inject(method = "<init>", at = @At("TAIL"))
	public void init(ResourcePackManager resourcePackManager, Consumer<ResourcePackManager> applier, Path file, Text title, CallbackInfo ci) {
		MinecraftClient.getInstance().options.resourcePacks = resourcePackManager.getEnabledProfiles().stream().filter(Predicate.not(ResourcePackProfile::isPinned)).map(ResourcePackProfile::getName).collect(Collectors.toCollection(ArrayList::new));;
	}
}
```
</details>

I kindly recommend considering the inclusion of this mixin in the mod to enhance its compatibility and provide a smoother resource pack selection screen experience for users.

Once again, I appreciate your efforts in maintaining this project and thank you for your understanding. Feel free to reach out if you have any questions or if there's anything I can assist you with.

Best regards,
Gerrygames

**Comment by MrKinau** [comment]
fixed in 1.20.5 rc1
